### PR TITLE
homework5

### PR DESCRIPTION
**Pull request by Karl5236543**

### CODE REVIEW

File: homework5.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#
+#2)Текстовый файл содержит записи о телефонах и их владельцах. Переписать в другой файл телефоны тех владельцев, фамилии которых начинаются с букв К и С.
+#
+#3) Получить файл, в котором текст выровнен по правому краю путем равномерного добавления пробелов.
+#
+#4)Дан текстовый файл со статистикой посещения сайта за неделю. Каждая строка содержит ip адрес, время 
+#и название дня недели (например, 139.18.150.126 23:12:44 sunday). Создайте новый текстовый файл, который 
+#бы содержал список ip без повторений из первого файла. Для каждого ip укажите количество посещений, 
+#наиболее популярный день недели. Последней строкой в файле добавьте наиболее популярный отрезок времени 
+#в сутках длиной один час в целом для сайта.
```
Comment (katevoskresenska) [inline homework5.py:11]: It is a general comment for all the tasks - the code should be represented with functions

File: homework5.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import csv
+from datetime import *
+
+#----------------------------------------------------------------------------------#
+# 1
+#----------------------------------------------------------------------------------#
+def task1():
+    new_lines = []
+    with open(r'.\homework5_data\1.txt', 'r', encoding='utf-8') as f:
```
Comment (mmykhailenko) [inline homework5.py:20]: check if file exists, before open it

File: homework5.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+#бы содержал список ip без повторений из первого файла. Для каждого ip укажите количество посещений, 
+#наиболее популярный день недели. Последней строкой в файле добавьте наиболее популярный отрезок времени 
+#в сутках длиной один час в целом для сайта.
+import csv
+from datetime import *
+
+#----------------------------------------------------------------------------------#
+# 1
+#----------------------------------------------------------------------------------#
+def task1():
```
Comment (mmykhailenko) [inline homework5.py:18]: name of the function is not quite understand

File: homework5.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    with open(r'.\homework5_data\1.txt', 'w', encoding='utf-8') as f:
+        f.writelines(new_lines)
+
+
+#----------------------------------------------------------------------------------#
+# 2
+#----------------------------------------------------------------------------------#
+def task2():
+    file_path = r'.\homework5_data\2_input.txt'
```
Comment (mmykhailenko) [inline homework5.py:38]: use this as an argument of function

File: homework5.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    required_phones = []
+
+    with open(r'.\homework5_data\2_input.txt', 'r', encoding='utf-8') as f:
```
Comment (mmykhailenko) [inline homework5.py:41]: check if file exist

File: homework5.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        required_phones = [line.split()[2] + '\n' for line in f if line.split()[1][0] in ['С', 'К']]
```
Comment (mmykhailenko) [inline homework5.py:42]: leave a comment here, what is that mean or divide by small blocks

File: homework5.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+#----------------------------------------------------------------------------------#
+# 4
+#----------------------------------------------------------------------------------#
+def task4():
+    input_file_path = r'.\homework5_data\4_input.csv'
+    output_file_path = r'.\homework5_data\4_output.txt'
+    output_data = []
+    
+    with open(input_file_path, newline='', encoding='utf-8') as csvfive:
+        reader = csv.DictReader(csvfive)
```
Comment (mmykhailenko) [inline homework5.py:73]: if you would like to be prepared to read csv files, then, probably would be a good idea to check the format of this file before this line

File: homework5.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        
+        times = []
+        time_list = [datetime.strptime(row['time'], "%H:%M:%S") for row in input_data]
+        for visit_time in time_list:
+            start = visit_time.time()
+            end = (visit_time + timedelta(hours=1)).time()
+            visits_count = len([row for row in time_list if start <= row.time() < end])
+            times.append((start, end, visits_count))
+        popular_time = max(times, key=lambda el : el[2])
+        output_data.append(f'Самый популярный период времени: {popular_time[0]} - {popular_time[1]}')
```
Comment (mmykhailenko) [inline homework5.py:90]: keep small manager of context. Try to out of with open block as fast as you can

**Review by mmykhailenko — APPROVED** [review]

**Pull request merged**